lib/plugins/aws/remove/lib/bucket.js

```diff
@@ -78,12 +79,18 @@ module.exports = {
 
   async deleteObjects() {
     if (this.objectsInBucket.length) {
-      await this.provider.request('S3', 'deleteObjects', {
+      const data = await this.provider.request('S3', 'deleteObjects', {
         Bucket: this.bucketName,
         Delete: {
           Objects: this.objectsInBucket,
         },
       });
+      if (data && data.Errors && data.Errors.length) {
```

Comment:
What do you think of exposing underlying error message to user? Not having the exact error info, may really complicate further debugging on user side

Reply:
I was thinking about it, but it will often be multiple errors (separate error for each s3 bucket object) and exposing all of them might produce a lot of noise - what do you think?

Reply:
I think the user needs to know what really happens.

If we're sure that error is about missing permissions then what we throw now is good. But if the error is about something else, we're sending the wrong information.

See e.g. how we handle such situations in other places: https://github.com/serverless/serverless/blob/409b0ed60f247e3803de0cb742967b7ba771c295/lib/aws/request.js#L186-L191

I'm not sure exactly what's the best approach here, as I didn't investigate what possible errors we can deal with here.

Reply:
I will think about how we can address it, because the errors potentially can be mixed, e.g. we can have error about access denied and potentially about something else. What do you think about checking the error code of the first (or all?) errors returned and if we see only those about access denied then we suggest that they are missing permissions? Otherwise, maybe we can inject the unique error codes into the error message?

Reply:
+1 on more details.

> What do you think about checking the error code of the first

Showing the content of the first error would probably cover 95% of all cases, that sounds reasonable to me (it's an edge case, it doesn't have to be as perfect as the nominal case). Also even if the error message gets a bit longer it will be a lot more useful if it contains the first error from S3.

And the CLI error can also mention that it's just the first S3 error message, e.g.

> Error while emptying the S3 deployment bucket (bucket-name). Make sure that you have permissions that allow S3 objects deletion.
> S3 error: &lt;message&gt;

In the extreme scenario where there are actually 2 different errors, then the user will fix the first error, try again and see the second error. It will still work, it's just 2 steps instead of 1, which sounds reasonable.

Reply:
I've changed the approach to inspect the first error and adjust the message a bit if it's `AccessDenied` to provide more actionable information. In both cases I included the error code of the first error. Let me know what do you think

Reply:
@pgrzesik Maybe consider logging those errors in verbose mode?

Reply:
Hey @bityob - the problem with logging all of them is that it will usually be 10 or more such errors and that can become too noisy - if the current approach won't be good enough then we will revisit and potentially improve it in the future.